releases/release-5.0.4.md

```diff
@@ -0,0 +1,165 @@
+---
+title: TiDB 5.0.4 Release Notes
+---
+
+# TiDB 5.0.4 Release Notes
+
+Release Date: September 27, 2021
+
+TiDB version: 5.0.4
+
+## Compatibility Changes
+
++ TiDB
+
+    - Revert #19341 to avoid #24326 [#26258](https://github.com/pingcap/tidb/pull/26258)
+
+## Improvements
+
++ TiDB
+
+    - Trigger auto-analyze based on histogram row count [#26707](https://github.com/pingcap/tidb/pull/26707)
```

Comment:
- TiDB
  - Trigger auto-analyze based on histogram row count [#26707](https://github.com/pingcap/tidb/pull/26707)
  - block the tiflash node for a period when it fails before. [#26757](https://github.com/pingcap/tidb/pull/26757)
  - Increasing the split region limit makes the split table and pre split more stable [#26657](https://github.com/pingcap/tidb/pull/26657)
  - support retry for mpp query [#26483](https://github.com/pingcap/tidb/pull/26483)
  - check the tiflash availabilities before launching mpp queries. [#26356](https://github.com/pingcap/tidb/pull/26356)
  - Fix copt-cache metrics, it will display the number of hits/miss/evict on Grafana. [#26343](https://github.com/pingcap/tidb/pull/26343)
  - Fix the issue that concurrently truncating the same partition hangs DDL. [#26238](https://github.com/pingcap/tidb/pull/26238)
  - support stable result mode [#26084](https://github.com/pingcap/tidb/pull/26084)
  - TiDB now supports the mysql system variable init_connect and associated functionality. [#26072](https://github.com/pingcap/tidb/pull/26072)
  - thoroughly push down count-distinct agg in the MPP mode. [#25861](https://github.com/pingcap/tidb/pull/25861)
  - Log warnings when agg function can not be pushdown in explain statement [#25736](https://github.com/pingcap/tidb/pull/25736)
  - Add err label for TiFlashQueryTotalCounter [#25327](https://github.com/pingcap/tidb/pull/25327)
  - add table name in log [#24802](https://github.com/pingcap/tidb/pull/24802)
  - Support getting the MVCC key of the secondary index in a clustered index table by HTTP API. [#24470](https://github.com/pingcap/tidb/pull/24470)
  - Optimize the parser object allocation in the prepared statement. [#24371](https://github.com/pingcap/tidb/pull/24371)

Reply:
Note: overwrite the original release notes in this PR.